### DISCUSSION TITLE
Select only a string column in association QueryExpressionMethod does not work

### DISCUSSION THREAD

**Discussion by timbze** [linq2db/linq2db#4721] [Q&A] [answered]
Similar to #1465

I tried to do basically this:
```c#
[Association(QueryExpressionMethod = nameof(OtherImpl), CanBeNull = true)]
public string? Other { get; set; }

private static Expression<Func<SomeEntity, IDataContext, IQueryable<string?>>> OtherImpl()
{
  return (e, db) => db.GetTable<SomeOtherEntity>().Where(se => se.Id == e.Id)
     .Select(o => o.StrValue)
     .Take(1); // though I did not add this Take()
}
```

But the sql that got generated had subquery with `select * from` and then did not work. I thought the subquery should have been `select StrValue from`.

Is this something I can do? Or is it not supported?

**Comment by MaceWindu** [comment]
Do you need it to be association to be able to load it explicitly with LoadWith?
if no, you can use expression method:
```cs
[ExpressionMethod(nameof(OtherImpl))]
public string? Other { get; set; }

private static Expression<Func<SomeEntity, IDataContext, string?>> OtherImpl()
{
	return (e, db) => db.GetTable<SomeOtherEntity>().Where(se => se.Id == e.Id)
		.Select(o => o.StrValue)
		.FirstOrDefault();
}
```

**Reply by timbze** [reply]
I will check it. I preferred being able to use it when I want because I don't need the extra join (or subquery) for all my queries...

**Reply by MaceWindu** [reply, marked as answer]
I've created issue #4723, we probably will add support for such mappings to next release
